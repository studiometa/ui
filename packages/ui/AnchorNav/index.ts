export * from './AnchorNav.js';
export * from './AnchorNavLink.js';
export * from './AnchorNavTarget.js';
