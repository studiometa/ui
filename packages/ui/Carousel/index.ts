export * from './Carousel.js';
export * from './CarouselBtn.js';
export * from './CarouselDrag.js';
export * from './CarouselItem.js';
export * from './CarouselProgress.js';
export * from './CarouselWrapper.js';
export * from './AbstractCarouselChild.js';
