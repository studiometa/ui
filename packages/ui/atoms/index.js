export * from './Figure/index.js';
export * from './Prefetch/index.js';
export * from './ScrollAnimation/index.js';
export * from './ScrollReveal/index.js';
export { default as AnchorScrollTo } from './AnchorScrollTo/AnchorScrollTo.js';
export { default as Cursor } from './Cursor/Cursor.js';
export { default as LargeText } from './LargeText/LargeText.js';
export { default as LazyInclude } from './LazyInclude/LazyInclude.js';
