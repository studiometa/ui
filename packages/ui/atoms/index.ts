export * from './AnchorScrollTo/index.js';
export * from './Cursor/index.js';
export * from './Figure/index.js';
export * from './LargeText/index.js';
export * from './LazyInclude/index.js';
export * from './Prefetch/index.js';
export * from './ScrollAnimation/index.js';
export * from './ScrollReveal/index.js';
