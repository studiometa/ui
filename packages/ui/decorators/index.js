export { default as withTransition } from './withTransition.js';
