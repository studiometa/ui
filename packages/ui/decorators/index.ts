export * from './withTransition.js';
export * from './withDeprecation.js';
export * from './withIndex.js';
