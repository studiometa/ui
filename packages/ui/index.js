export * from './primitives/index.js';
export * from './atoms/index.js';
export * from './molecules/index.js';
export * from './organisms/index.js';
export * from './decorators/index.js';
