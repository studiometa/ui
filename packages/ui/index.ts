export * from './Accordion/index.js';
export * from './Action/index.js';
export * from './AnchorNav/index.js';
export * from './AnchorScrollTo/index.js';
export * from './Carousel/index.js';
export * from './CircularMarquee/index.js';
export * from './Cursor/index.js';
export * from './Data/index.js';
export * from './decorators/index.js';
export * from './Draggable/index.js';
export * from './Figure/index.js';
export * from './FigureVideo/index.js';
export * from './Frame/index.js';
export * from './Indexable/index.js';
export * from './Hoverable/index.js';
export * from './LargeText/index.js';
export * from './LazyInclude/index.js';
export * from './Menu/index.js';
export * from './Modal/index.js';
export * from './Panel/index.js';
export * from './Prefetch/index.js';
export * from './ScrollAnimation/index.js';
export * from './ScrollReveal/index.js';
export * from './Sentinel/index.js';
export * from './Slider/index.js';
export * from './Sticky/index.js';
export * from './Tabs/index.js';
export * from './Transition/index.js';
