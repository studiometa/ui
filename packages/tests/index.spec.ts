import { test, expect } from 'vitest';
import * as components from '@studiometa/ui';

test('components exports', () => {
  expect(Object.keys(components).toSorted()).toMatchInlineSnapshot(`
    [
      "AbstractCarouselChild",
      "AbstractFrameTrigger",
      "AbstractPrefetch",
      "AbstractScrollAnimation",
      "AbstractSliderChild",
      "Accordion",
      "AccordionItem",
      "Action",
      "AnchorNav",
      "AnchorNavLink",
      "AnchorNavTarget",
      "AnchorScrollTo",
      "Carousel",
      "CarouselBtn",
      "CarouselDrag",
      "CarouselItem",
      "CarouselWrapper",
      "CircularMarquee",
      "Cursor",
      "DataBind",
      "DataComputed",
      "DataEffect",
      "DataModel",
      "Draggable",
      "Figure",
      "FigureShopify",
      "FigureTwicpics",
      "FigureVideo",
      "FigureVideoTwicpics",
      "Frame",
      "FrameAnchor",
      "FrameForm",
      "FrameLoader",
      "FrameTarget",
      "FrameTriggerLoader",
      "Hoverable",
      "Indexable",
      "LargeText",
      "LazyInclude",
      "Menu",
      "MenuBtn",
      "MenuList",
      "Modal",
      "ModalWithTransition",
      "Panel",
      "PrefetchWhenOver",
      "PrefetchWhenVisible",
      "ScrollAnimation",
      "ScrollAnimationChild",
      "ScrollAnimationChildWithEase",
      "ScrollAnimationParent",
      "ScrollAnimationWithEase",
      "ScrollReveal",
      "Sentinel",
      "Slider",
      "SliderBtn",
      "SliderCount",
      "SliderDots",
      "SliderDrag",
      "SliderItem",
      "SliderProgress",
      "Sticky",
      "Tabs",
      "Target",
      "Transition",
      "animationScrollWithEase",
      "withDeprecation",
      "withIndex",
      "withTransition",
    ]
  `);
});
