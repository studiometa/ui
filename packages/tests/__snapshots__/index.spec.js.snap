// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components exports 1`] = `
[
  "AbstractPrefetch",
  "AbstractScrollAnimation",
  "AbstractSliderChild",
  "Accordion",
  "AccordionItem",
  "AnchorScrollTo",
  "Cursor",
  "Draggable",
  "Figure",
  "FigureTwicpics",
  "Frame",
  "FrameAnchor",
  "FrameForm",
  "FrameTarget",
  "LargeText",
  "LazyInclude",
  "Menu",
  "MenuBtn",
  "MenuList",
  "Modal",
  "ModalWithTransition",
  "Panel",
  "PrefetchWhenOver",
  "PrefetchWhenVisible",
  "ScrollAnimation",
  "ScrollAnimationChild",
  "ScrollAnimationChildWithEase",
  "ScrollAnimationParent",
  "ScrollAnimationWithEase",
  "ScrollReveal",
  "Sentinel",
  "Slider",
  "SliderBtn",
  "SliderCount",
  "SliderDots",
  "SliderDrag",
  "SliderItem",
  "SliderProgress",
  "Sticky",
  "TableOfContent",
  "TableOfContentAnchor",
  "Tabs",
  "Transition",
  "animationScrollWithEase",
  "loadImage",
  "withTransition",
]
`;
